### PR TITLE
refactor: remove explicit handling of alignment

### PR DESCRIPTION
**Pull request by zvolin**

### CODE REVIEW

File: src/utils.rs [hunk 1]
```diff
@@ -20,20 +20,18 @@ pub fn is_aligned(shards: &[&[u8]]) -> bool {
...
+pub fn alloc_aligned(shards: usize, shard_size: usize) -> Result<Vec<u8>> {
+    if shard_size % ALIGN != 0 {
+        return Err(LeopardError::InvalidShardSize(shard_size));
     }
 
-    let ratio = SHARD_SIZE / ALIGN;
-
     // create a vec of Aligned to make sure they are aligned properly
-    let aligned_mem = vec![Aligned::zeroed(); shards * ratio];
+    let aligned_mem = vec![Aligned::zeroed(); shards * shard_size / ALIGN];
```
Comment (oblique) [inline src/utils.rs]: Any reasons on why you did't use [Layout::from_size_align](https://doc.rust-lang.org/stable/std/alloc/struct.Layout.html#method.from_size_align) + [alloc](https://doc.rust-lang.org/stable/std/alloc/fn.alloc.html)

Comment (zvolin) [reply src/utils.rs]: likely just because I wasn't aware of it :+1:

Comment (zvolin) [reply src/utils.rs]: so this as well as prev attempt turned out to be unsound, violating rules in `Vec::from_raw_parts`. refactored whole thing

**Review by oblique — APPROVED** [review]
lgtm

**Review by oblique — APPROVED** [review]
lgtm

**Review by fl0rek — APPROVED** [review]
🚀

**Review by oblique — APPROVED** [review]

**Pull request merged**